lib/Site.js

```diff
@@ -393,7 +415,15 @@ Site.prototype.regenerateAffectedPages = function (filePaths) {
 
   return new Promise((resolve, reject) => {
     Promise.all(processingFiles)
-      .then(() => logger.info('Pages rebuilt'))
+      .then((updatedFrontMatter) => {
+        logger.info('Pages rebuilt');
+        updatedFrontMatter.forEach((currentData) => {
```

Comment:
As mentioned in `Page.prototype.generate`, `page.generate` shouldn't return `updatedFrontMatter`.
You should be able to get it from each `page` in `this.pageModels`.

```js
this.pageModels.forEach((page) => {
  const currentData = page.frontMatter;
  // ...
});
```

Then revert:

```diff
- .then(() => logger.info('Pages rebuilt'))
```